src/AppContainer.react.js

```diff
@@ -21,8 +21,8 @@ class UnconnectedAppContainer extends React.Component {
     }
 
     componentWillMount() {
-        const {dispatch} = this.props;
-        dispatch(readConfig());
+        const {dispatch, initialConfig} = this.props;
+        dispatch(readConfig({config: initialConfig}));
```

Comment:
Let's move `|| JSON.parse(document.getElementById('_dash-config').textContent)` up here. We never should have put that in the reducer in the first place, but with the extra logic it's even more out of place.

Also the extra nesting is unnecessary, can't we just do `readConfig(initialConfig || ...)`?